### PR TITLE
Adding tests to cover Batch request overflow

### PR DESCRIPTION
**Pull request by ealsur**
## Description

As part of the work for https://github.com/Azure/azure-cosmos-dotnet-v3/issues/584, this PR adds new tests to cover possible batch request overflow due to materialization.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Batch/PartitionKeyRangeServerBatchRequestTests.cs [hunk 1]
```diff
@@ -59,5 +59,73 @@ public async Task OverflowsBasedOnCount()
...
             Assert.AreEqual(operations[2].Id, pendingOperations[1].Id);
         }
+
+        [TestMethod]
+        public async Task PartitionKeyRangeServerBatchRequestSizeTests()
+        {
+            const int docSizeInBytes = 250;
+            const int operationCount = 10;
+
+            foreach (int expectedRequestCount in new int[] { 1, 2, 5, 10 })
```
Comment (abhijitpai) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Batch/PartitionKeyRangeServerBatchRequestTests.cs]: Please make this expectedOperationCount - in the earlier tests, they were supposed to be count of batch requests produced, but here they represent the number of operations in the request produced.

File: Microsoft.Azure.Cosmos/src/Batch/ServerBatchRequest.cs [hunk 2]
```diff
@@ -134,7 +134,7 @@ public MemoryStream TransferBodyStream()
                 throw new RequestEntityTooLargeException(RMResources.RequestTooLarge);
             }
 
-            return new ArraySegment<ItemBatchOperation>(operations.Array, materializedCount, operations.Count - materializedCount);
+            return new ArraySegment<ItemBatchOperation>(operations.Array, this.operations.Count, operations.Count - this.operations.Count);
```
Comment (abhijitpai) [inline Microsoft.Azure.Cosmos/src/Batch/ServerBatchRequest.cs]: Does this assume that the input array segment was initialized with offset 0? Am asking if operations.Offset needs to be added to this.operations,count.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Batch/ServerBatchRequest.cs]: Good catch, yes, Offset should be taken into account.
